tests/configuration_tests.py

```diff
@@ -0,0 +1,119 @@
+                'test event 1': [
+                    dict(name="Reused Name", arg="reused_arg", model="auth.User"),
+                    dict(name="Reused Name", arg="reused_arg", model="auth.User"),
+                    dict(name="foobar", arg="", model="auth.User"),
+                    dict(name="", arg="invalid arg", model="auth.User"),
+                    dict(name="Not Reused Name", arg="not_reused_arg",
+                         model="auth.User"),
+                ],
+                'test event 2': [
+                    dict(name="Not Reused Name", arg="not_reused_arg",
+                         model="auth.User"),
+                    dict(foobar="abc"),
+                ],
+                'test event 3': [
+                    dict(name="User1", arg="user1", model="invalid_app.User"),
+                    dict(name="User2", arg="user2", model="auth.InvalidModel"),
+                ],
+            },
+        }
+
+        util.validate_configuration(config)
+
+        expected_messages = set([
+            "invalid event name: ''",
+            "reused data name for event 'test event 1': 'Reused Name'",
+            "reused data argument for event 'test event 1': 'reused_arg'",
+            "invalid data argument for event 'test event 1': ''",
+            "invalid data argument for event 'test event 1': 'invalid arg'",
+            "invalid data name for event 'test event 1': ''",
+            "unexpected data key for event 'test event 2': 'foobar'",
+            "missing data key for event 'test event 2': 'name'",
+            "missing data key for event 'test event 2': 'arg'",
+            "missing data key for event 'test event 2': 'model'",
+            ("bad model 'invalid_app.User' for event 'test event 3': "
+             "No installed app with label 'invalid_app'."),
+            ("bad model 'auth.InvalidModel' for event 'test event 3': "
+             "App 'auth' doesn't have a 'invalidmodel' model."),
+        ])
+        actual_messages = [message.getMessage()
+                           for message in handler.messages]
+
+        for message in actual_messages:
+            if message not in expected_messages:
+                raise self.failureException("got unexpected log message '{}'"
+                                            .format(message))
+
+            expected_messages.remove(message)
+
+        if expected_messages:
+            raise self.failureException(
+                "expected log message(s) missing: {}"
+                .format(list(expected_messages)))
+
+    @override_settings()
+    @mock.patch('django_emarsys.util.validate_configuration')
+    def test_validate_settings(self, mock_validate_configuration):
+        expected_config = {key: getattr(settings, key)
+                           for key in dir(settings)
+                           if hasattr(settings, key)}
+        util.validate_settings()
+
+        mock_validate_configuration.assert_called_with(expected_config)
```

Comment:
What's the point of this test?

The benefit of a test is correlated with the probability that the test catches a bug in the implementation, and inversely correlated with my confidence that the tested code works (without testing it). (That's why simpler code is better - I can be more confident that it works correctly if it's easier to reason about it.)

The cost of a test is that it has to be maintained, adds running time to the test suite, and may be buggy itself.

In this case the test code duplicates the function's implementation. That reduces the benefit to zero - if I have a bug in the implementation, it will be in the test as well.